src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -579,42 +621,62 @@ void ReplicatedMergeTreeSink::commitPart(
                 storage.unlockSharedData(*part);
                 /// NOTE: We could be here if the node with the quorum existed, but was quickly removed.
                 transaction.rollback();
-                throw Exception("Unexpected logical error while adding block " + toString(block_number) + " with ID '" + block_id + "': "
-                                + Coordination::errorMessage(multi_code) + ", path " + failed_op_path,
-                                ErrorCodes::UNEXPECTED_ZOOKEEPER_ERROR);
+                throw Exception(
+                    ErrorCodes::UNEXPECTED_ZOOKEEPER_ERROR,
+                    "Unexpected logical error while adding block {} with ID '{}': {}, path {}",
```

Comment:
Do I understand correctly, that we will not retry on this error? (It's probably ok, because the error message looks frightening and probably it's not retryable, just want to clarify)

Reply:
Maybe we should try to replace error code with `LOGICAL_ERROR` (so it will crash in debug build)

Reply:
Here I just reformat the message. It's not a connectivity error with zookeeper, so I don't really want to touch it here